theme/oauth-authorize.ftl

```diff
@@ -0,0 +1,111 @@
+          ${theme.message('pending-links-login-to-complete', devicePendingIdPLink.identityProviderName, pendingIdPLink.identityProviderName)}
+        [#elseif devicePendingIdPLink??]
+          ${theme.message('pending-link-login-to-complete', devicePendingIdPLink.identityProviderName)}
+        [#else]
+          ${theme.message('pending-link-login-to-complete', pendingIdPLink.identityProviderName)}
+        [/#if]
+        [#-- A pending link can be cancled. If we also have a device link in progress, this cannot be canceled. --]
+        [#if pendingIdPLink??]
+          [@helpers.link url="" extraParameters="&cancelPendingIdpLink=true"]${theme.message('login-cancel-link')}[/@helpers.link]
+        [/#if]
+        </p>
+      [/#if]
+      <form action="${request.contextPath}/oauth2/authorize" method="POST" class="full">
+        <h1 class="form-title">Authorize Application</h1>
+        [@helpers.oauthHiddenFields/]
+        [@helpers.hidden name="showPasswordField"/]
+        [#if showPasswordField && hasDomainBasedIdentityProviders]
+          [@helpers.hidden name="loginId"/]
+        [/#if]
+
+        [@helpers.prInput type="text" fieldName="loginId" autocomplete="username" placeholder=theme.message('loginId') autofocus=true required=true disabled=(showPasswordField && hasDomainBasedIdentityProviders) /]
+        [#if showPasswordField]
+          [@helpers.prInput type="password" fieldName="password" autocomplete="current-password" autofocus=loginId?has_content placeholder=theme.message('password') /]
+          <!-- [@helpers.captchaBadge showCaptcha=showCaptcha captchaMethod=tenant.captchaConfiguration.captchaMethod siteKey=tenant.captchaConfiguration.siteKey/] -->
+        [/#if]
+
+        <div class="form-check">
+          [@helpers.prCheckbox fieldName="rememberDevice" label=theme.message('remember-device') value="true" uncheckedValue="false"]
+            <i class="fa fa-info-circle" data-tooltip="${theme.message('{tooltip}remember-device')}"></i>[#t/]
+          [/@helpers.prCheckbox]
+        </div>
+
+          <div class="form-row">
+            [#if showPasswordField]
+              <button type="submit" class="btn btn-primary">${theme.message('submit')}</button>
+              <br />
+              <div style="text-align: center">
+                [@helpers.link url="${request.contextPath}/password/forgot"]${theme.message('forgot-your-password')}[/@helpers.link]
+              </div>
+            [#else]
+              <button type="submit" class="btn btn-primary">${theme.message('next')}</button>
+            [/#if]
+          </div>
+      </form>
+      <div style="text-align: center">
+        [#if showPasswordField && hasDomainBasedIdentityProviders]
+          [@helpers.link url="" extraParameters="&showPasswordField=false"]${theme.message('sign-in-as-different-user')}[/@helpers.link]
+        [/#if]
+      </div>
+      [#if application.registrationConfiguration.enabled]
```

Comment:
Hm... I think this is probably the right call, but wanted to at least think out loud here for a moment.

The existing login page has a link to the signup page. This FusionAuth-hosted login page is currently used by the mobile apps, which handle signup in-app. We are (probably?) holding off on using this login page until we have FusionAuth-hosted signups too, in which case we'll (probably?) enable registration and then this will become true. So, the implemented behavior is correct: we shouldn't show a link to the signup page here.

Right?